### PR TITLE
Restart on update

### PR DESCRIPTION
**Pull request by alexjch**
Because telemd does not restart when an update is installed the daemon could get out of sync with probes if a new version of a probe requires a new version of telemd. This change will trigger a restart when an update takes place.

### CODE REVIEW

File: configure.ac [hunk 1]
```diff
@@ -93,11 +93,18 @@ test -z "${tmpfilespath}" && tmpfilespath=/usr/lib/tmpfiles.d
 AC_SUBST(SYSTEMD_TMPFILESDIR, [${tmpfilespath}])
 
 AC_ARG_WITH([systemdsysctldir], AS_HELP_STRING([--with-systemdsysctldir=DIR],
-	    [path to systemd sysctl dir @<:@default=/usr/lib/sysctl.d@:>@]), [sysctlpath=${withval}],
-	    [sysctlpath="$($PKG_CONFIG --variable=sysctldir systemd)"])
+	    [path to systemd sysctl dir @<:@default=/usr/lib/sysctl.d@:>@]), [sysctlpath=${withval}])
```
Comment (phmccarty) [inline configure.ac]: This change looks unrelated to the rest of your patch.

File: configure.ac [hunk 2]
```diff
@@ -93,11 +93,18 @@ test -z "${tmpfilespath}" && tmpfilespath=/usr/lib/tmpfiles.d
...
 test -z "${sysctlpath}" && sysctlpath=/usr/lib/sysctl.d
 AC_SUBST(SYSTEMD_SYSCTLDIR, [${sysctlpath}])
 
+
+AC_ARG_WITH([systemdsystemctldir], AS_HELP_STRING([--with-systemdsystemctldir=DIR],
+            [path to systemd systemctl dir @<:@default=/usr/bin@:>@]), [systemctlpath=${withval}],
+            [systemctlpath="$($PKG_CONFIG --variable=systemctldir systemd)"])
```
Comment (phmccarty) [inline configure.ac]: The `systemctldir` variable does not exist in systemd.pc, so you can omit the fourth argument to this macro.

**Comment by phmccarty** [comment]
@alexjch Can you also add a commit description to describe the reasoning behind this change?

File: configure.ac (hunk 1 above)
Comment (alexjch) [reply configure.ac]: Yep, I messed up my patch.

File: configure.ac [hunk 3]
```diff
@@ -93,11 +93,16 @@ test -z "${tmpfilespath}" && tmpfilespath=/usr/lib/tmpfiles.d
 AC_SUBST(SYSTEMD_TMPFILESDIR, [${tmpfilespath}])
 
 AC_ARG_WITH([systemdsysctldir], AS_HELP_STRING([--with-systemdsysctldir=DIR],
-	    [path to systemd sysctl dir @<:@default=/usr/lib/sysctl.d@:>@]), [sysctlpath=${withval}],
-	    [sysctlpath="$($PKG_CONFIG --variable=sysctldir systemd)"])
+            [path to systemd sysctl dir @<:@default=/usr/lib/sysctl.d@:>@]), [sysctlpath=${withval}],
+            [sysctlpath="$($PKG_CONFIG --variable=sysctldir systemd)"])
```
Comment (phmccarty) [inline configure.ac]: whitespace changes... Looks like the rest of the file uses tabs+spaces, so I would switch it back for consistency.

Comment (alexjch) [reply configure.ac]: lovely... I'll fix

**Comment by phmccarty** [comment]
+1. Do you want me to squash the commits and use your PR description for the single commit?

**Comment by phmccarty** [comment]
Okay, I squashed the three commits into one, and merged to master. Thanks!
